partiql-eval/src/main/kotlin/org/partiql/eval/internal/operator/rex/ExprCast.kt

```diff
@@ -1,18 +1,354 @@
 package org.partiql.eval.internal.operator.rex
 
+import com.amazon.ion.Decimal
+import com.amazon.ionelement.api.ElementType
+import com.amazon.ionelement.api.IonElementException
+import com.amazon.ionelement.api.createIonElementLoader
+import org.partiql.errors.DataException
+import org.partiql.errors.TypeCheckException
 import org.partiql.eval.internal.Record
 import org.partiql.eval.internal.operator.Operator
 import org.partiql.plan.Ref
+import org.partiql.value.BagValue
+import org.partiql.value.BoolValue
+import org.partiql.value.CollectionValue
+import org.partiql.value.DecimalValue
+import org.partiql.value.Float32Value
+import org.partiql.value.Float64Value
+import org.partiql.value.Int16Value
+import org.partiql.value.Int32Value
+import org.partiql.value.Int64Value
+import org.partiql.value.Int8Value
+import org.partiql.value.IntValue
+import org.partiql.value.ListValue
+import org.partiql.value.NumericValue
 import org.partiql.value.PartiQLValue
 import org.partiql.value.PartiQLValueExperimental
+import org.partiql.value.PartiQLValueType
+import org.partiql.value.SexpValue
+import org.partiql.value.StringValue
+import org.partiql.value.SymbolValue
+import org.partiql.value.TextValue
+import org.partiql.value.bagValue
+import org.partiql.value.boolValue
+import org.partiql.value.decimalValue
+import org.partiql.value.float32Value
+import org.partiql.value.float64Value
+import org.partiql.value.int16Value
+import org.partiql.value.int32Value
+import org.partiql.value.int64Value
+import org.partiql.value.int8Value
+import org.partiql.value.intValue
+import org.partiql.value.listValue
+import org.partiql.value.sexpValue
+import org.partiql.value.stringValue
+import org.partiql.value.symbolValue
+import java.math.BigDecimal
+import java.math.BigInteger
 
-internal class ExprCast(
-    private val arg: Operator.Expr,
-    private val cast: Ref.Cast,
-) : Operator.Expr {
-
+// TODO: This is incomplete
+internal class ExprCast(val arg: Operator.Expr, val cast: Ref.Cast) : Operator.Expr {
     @OptIn(PartiQLValueExperimental::class)
     override fun eval(record: Record): PartiQLValue {
-        TODO("Not yet implemented")
+        val arg = arg.eval(record)
+        try {
+            return when (arg.type) {
```

Comment:
This is ok for now, but considering the planner knows the argument input type we can actually select a branch at compile-time rather than runtime.